cmd/gateway/azure/gateway-azure.go

```diff
@@ -505,11 +512,116 @@ func (a *azureObjects) DeleteBucket(ctx context.Context, bucket string) error {
 	return azureToObjectError(err, bucket)
 }
 
+// Filter out the entries which are lexicographically <= marker and return the rest of the listing.
+func (a *azureObjects) filterListObjects(ctx context.Context, bucket, prefix, marker, delimiter string, maxKeys int) (result minio.ListObjectsInfo, err error) {
+	var objects []minio.ObjectInfo
+	var prefixes []string
+	container := a.client.GetContainerReference(bucket)
+	azureListMarker := ""
+	for {
+		resp, err := container.ListBlobs(storage.ListBlobsParameters{
+			Prefix:     prefix,
+			Marker:     azureListMarker,
+			Delimiter:  delimiter,
+			MaxResults: uint(maxKeys),
+		})
+		if err != nil {
+			logger.LogIf(ctx, err)
+			return result, azureToObjectError(err, bucket, prefix)
+		}
+		azureListMarker = resp.NextMarker
+		i := 0
+		// Skip entries <= marker
+		for ; i < len(resp.Blobs); i++ {
+			object := resp.Blobs[i]
+			if strings.HasPrefix(object.Name, minio.GatewayMinioSysTmp) {
+				continue
+			}
+			if object.Name <= marker {
+				continue
+			}
+			break
+		}
+		// All entries >= marker are returned to the client.
+		for _, object := range resp.Blobs[i:] {
```

Comment:
I see this pattern of `for` loops repeated also in `ListObjects()`.  Not sure that can be avoided by having higher level functions.